### DISCUSSION TITLE
Howto query in GraphQL for entries related to categories based on selection of categories in an entry?

### DISCUSSION THREAD

**Discussion by ctigelaar** [craftcms/cms#9849] [Q&A] [answered]
Can anybody tell me how to query in GraphQL the entries related to a category, from an entry with some of those categories selected?

So I have for example:
- a category group with some drink categories, i.e. "wine", "beer", "soda".
- Collection of entries with some wines, beers, sodas categorised by those categories.
- Article about beer (slug: `more-beer`), where I select the category "beer" in a Category field `productCategories` to show some beer products below the content.

I'm querying for the article with an `entry(section: "articles", slug: "more-beer")`, but then I also want to get the entries related to the the selected category "beer". In other words the **entries related to the selected categories**, not just the selected **categories itself**.

Because this gives me only the categories, but I want the entries in those categories:
```gql
entry(section: "articles", slug: "more-beer") {
  id
  title
  uri
  ... on articles_Entry {
    productCategories {
      id
      title
    }
  }
}
```

Can't figure it out, tried some things with `relatedToEntries`, `relatedToCategories` and even separate `entries()` query. Without any luck so far. Anybody here who can help me with this? Think it should be possible out of the box without fancy plugins or multiple queries, not? Thanks.

**Comment by andris-sevcenko** [comment, marked as answer]
You would need to run a separate query for that. The `entry` query for the main entry to display and the other one to get the entries related to the category.

Something like 
```
entries (relatedToCategories: [{slug: "beer"}]) {
  title
  uri
  id
}
```

should work.

**Reply by ctigelaar** [reply]
Thanks Andris, so there is no way to re-use the already found category id's within the same entry query? Was hoping for a single query to solve this, but seems like it's not possible then. Is it something that will/can be added in the future?

**Reply by andris-sevcenko** [reply]
If I'm understanding correctly you mean fetching an entry and its related (via a category field) entries in a single go?

If so, that's currently not possible in Craft with a single query, since the entries are not directly related, but are related through a category field - you have to get the relevant categories first so you can query entries related to them.

We're looking in improving this in Craft 4.0.

**Reply by wbrowar** [reply]
Just like in #8954, I'm looking forward to this improvement!
